atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/AtlasClientFactory.java

```diff
@@ -0,0 +1,135 @@
+import com.palantir.common.proxy.ReplaceIfExceptionMatchingProxy;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.client.config.ClientConfiguration;
+import com.palantir.conjure.java.client.jaxrs.JaxRsClient;
+import com.palantir.conjure.java.okhttp.HostEventsSink;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.tritium.Tritium;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+public class AtlasClientFactory implements AutoCloseable {
+    static final StaticClientConfiguration EMPTY_STATIC_CONFIG =
+            StaticClientConfiguration.builder().build();
+
+    private final boolean useDialogueJaxrs;
+    private final Refreshable<Optional<ServerListConfig>> clientConfig;
+    private final TaggedMetricRegistry taggedMetricRegistry;
+    private final UserAgent userAgent;
+    private final HostEventsSink hostEventsSink;
+    private final DialogueClientFactory dialogueClientFactory;
+
+    public AtlasClientFactory(
+            Refreshable<ServerListConfig> clientConfig,
+            TaggedMetricRegistry taggedMetricRegistry,
+            UserAgent userAgent,
+            HostEventsSink hostEventsSink,
+            boolean useDialogueJaxrs) {
+        this.useDialogueJaxrs = useDialogueJaxrs;
+        this.clientConfig = clientConfig.map(c -> c.hasAtLeastOneServer() ? Optional.of(c) : Optional.empty());
+        this.taggedMetricRegistry = taggedMetricRegistry;
+        this.userAgent = userAgent;
+        this.hostEventsSink = hostEventsSink;
+        this.dialogueClientFactory = new DialogueClientFactory(this.clientConfig, taggedMetricRegistry, userAgent);
+    }
+
+    public <T> T client(Class<T> serviceClass) {
+        return client(serviceClass, EMPTY_STATIC_CONFIG);
+    }
+
+    // TODO(forozco): support multiple services
+    public <T> T client(Class<T> serviceClass, StaticClientConfiguration staticConfig) {
+        SafeArg<String> serviceClassSafeArg = SafeArg.of("serviceClass", serviceClass.getName());
+        Preconditions.checkArgument(
+                serviceClass.isInterface(), "serviceClass must be an interface", serviceClassSafeArg);
+
+        if (DialogueClientFactory.isDialogue(serviceClass)) {
+            return dialogueClientFactory.dialogueClient(serviceClass, staticConfig);
+        } else if (useDialogueJaxrs) {
+            return dialogueClientFactory.jaxrsClient(serviceClass, staticConfig);
+        }
+        return jaxrsInternal(serviceClass, staticConfig);
+    }
+
+    private <T> T jaxrsInternal(Class<T> serviceClass, StaticClientConfiguration staticClientConfig) {
+        Function<ServerListConfig, T> clientFactory = jaxrsFactory(serviceClass,
+                staticClientConfig);
+        Supplier<Optional<T>> client = clientConfig.map(maybeConf ->
```

Comment:
Can you explain why you couldn't use the AtlasDbHttpClients codepath? It's not just ReplaceIfExceptionMatchingProxy that's there, there's also FastFailoverProxy that you need to wire up. I suspect it also needs to be wired up in the Dialogue version.

Reply:
Changed it over. We should have this whole codepath be either on or off.